### PR TITLE
Tarmac plotting

### PR DESCRIPTION
**Pull request by sndrtj**

### CODE REVIEW

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala [hunk 1]
```diff
@@ -167,6 +167,76 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
...
+  case class SortBgzipTabix(sortJob: BedtoolsSort, bgzipJob: Bgzip, tabixJob: Tabix)
+
+  def sortBgzipAndTabix(file: File): SortBgzipTabix = {
+    val sorter = new BedtoolsSort(this)
+    sorter.input = file
+    sorter.isIntermediate = true
+    val sortFile = swapExt(file.getParentFile, file, ".bed", ".sorted.bed")
+    sorter.output = sortFile
+    val gzFile = swapExt(file.getParentFile, file, ".bed", ".sorted.bed.gz")
+    val bgzip = new Bgzip(this)
```
Comment (ffinfo) [inline tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala:245]: Is it possible to combine the Sort job and the Bgzip job?

File: tarmac/src/main/resources/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/tarmac_plot.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from os import makedirs
+import sys
+
+
+def get_middle_pos(record):
+    x = (int(record[2]) - int(record[1])) // 2
+    return int(record[1]) + x
+
+
+def plot_call(chrom, start, end, whandle, xhandle, shandle, margin, output_loc):
```
Comment (ffinfo) [inline tarmac/src/main/resources/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/tarmac_plot.py:36]: Is this plot somehow tested? If not this should be the next task in functional tests

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/TarmacPlot.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    getPythonCommand +
+      required("-c", callFile) +
+      required("-w", wisecondorFile) +
+      required("-s", stouffFile) +
+      required("-x", xhmmFile) +
+      required("-m", margin) +
+      required("-o", outputDir)
+  }
+
+}
```
Comment (ffinfo) [inline tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/TarmacPlot.scala:42]: It look like the new style reformatter is not in there yet. Could you update this branch to the latest develop? Or first update `feature-tarmac-pipeline` branch for this.

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala [hunk 4]
```diff
@@ -167,6 +167,76 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
     addAll(zScoreMergeJobs.values)
     addAll(windowStouffJobs.values.flatMap(_.values))
     addAll(thresholdJobs.values.flatMap(_.values))
+
+    val xhmmZGzip = xhmmZJobs map {
+      case (sample, jobAndRefFile) =>
+        val sort = sortBgzipAndTabix(jobAndRefFile._2)
+        val pipeJob = new BiopetFifoPipe(this, sort.sortJob :: sort.bgzipJob :: Nil)
+        add(sort.tabixJob)
+        add(pipeJob)
```
Comment (ffinfo) [inline tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: You can also write this as add(pipeJob, sort.tabixJob)

**Review by ffinfo — CHANGES_REQUESTED** [review]

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala (hunk 1 above)
Comment (sndrtj) [reply tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala:245]: This is done in all cases this function is called. The reason not do this in this function is that `pipeJobs` in a `BiopetFifoPipe` is only populated _after_ its `beforeGraph` is called, and I want to access its constituent jobs before that time.

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala (hunk 4 above)
Comment (sndrtj) [reply tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: Fixed

File: tarmac/src/main/resources/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/tarmac_plot.py (hunk 2 above)
Comment (sndrtj) [reply tarmac/src/main/resources/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/tarmac_plot.py:36]: Yes, that's the next step.

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/TarmacPlot.scala (hunk 3 above)
Comment (sndrtj) [reply tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/scripts/TarmacPlot.scala:42]: Done

**Review by ffinfo — APPROVED** [review]

**Pull request merged**